src/commonMain/kotlin/fr/acinq/eclair/db/InMemoryPaymentsDb.kt

```diff
@@ -51,22 +51,25 @@ class InMemoryPaymentsDb : PaymentsDb {
         return outgoing[id]?.let { payment ->
             val parts = outgoingParts.values.filter { it.first == payment.id }.map { it.second }
             return when (payment.status) {
-                is OutgoingPayment.Status.Succeeded -> payment.copy(parts = parts.filter { it.status is OutgoingPayment.Part.Status.Succeeded })
+                // This doesn't work. Is there something like "instanceof" to check inheritance ???
+            //  is OutgoingPayment.Status.Completed.Succeeded -> {
+            //      payment.copy(parts = parts.filter { it.status is OutgoingPayment.Part.Status.Succeeded })
+            //  }
+                is OutgoingPayment.Status.Completed.Succeeded.OffChain -> {
```

Comment:
I thought it should work too. But it doesn't ...

As the code is now, it will pass all unit tests. If you change the code to what it was before, it will fail multiple unit tests:
```kotlin
return when (payment.status) {
    // This doesn't work ...
    is OutgoingPayment.Status.Completed.Succeeded -> {
        payment.copy(parts = parts.filter { it.status is OutgoingPayment.Part.Status.Succeeded })
    }
    else -> payment.copy(parts = parts)
}
```

For example, this test will fail:
OutgoingPaymentHandlerTestsCommon: 689: partial failure then fulfill (spec violation)

Reply:
Done in cc9e868 :hand_over_mouth: